File: package/main/src/Tool/escapeRegExp.ts
/**
 * Escapes characters in a string for use in a regular expression.
 * @param {string} string - The string to escape.
 * @returns {string} The escaped string.
 * @example
 * escapeRegExp("a.b"); // "a\\.b"
 */
export const escapeRegExp = (string: string): string => {
  return string.replaceAll(/[.*+?^${}()|[\]\\]/g, String.raw`\$&`);
};

File: package/umt_python/src/tool/escape_regexp.py
import re


def escape_regexp(string: str) -> str:
    """
    Escapes characters in a string for use in a regular expression.

    Args:
        string: The string to escape.

    Returns:
        The escaped string.

    Example:
        >>> escape_regexp("a.b")
        "a\\.b"
    """
    return re.sub(r"[$()*+.?\[\\\]^{|}]", r"\\\g<0>", string)
